codec-haproxy/src/main/java/io/netty/handler/codec/haproxy/HAProxyMessage.java

```diff
@@ -519,4 +516,40 @@ public int destinationPort() {
     public List<HAProxyTLV> tlvs() {
         return tlvs;
     }
+
+    @Override
+    public HAProxyMessage touch() {
+        return (HAProxyMessage) super.touch();
+    }
+
+    @Override
+    public HAProxyMessage touch(Object hint) {
+        if (leak != null) {
+            leak.record(hint);
+        }
+        return this;
+    }
+
+    @Override
+    public HAProxyMessage retain() {
+        return (HAProxyMessage) super.retain();
+    }
+
+    @Override
+    public HAProxyMessage retain(int increment) {
+        return (HAProxyMessage) super.retain(increment);
+    }
+
+    @Override
+    protected void deallocate() {
+        for (HAProxyTLV tlv : tlvs) {
+            tlv.release();
+        }
+
+        final ResourceLeakTracker<HAProxyMessage> leak = this.leak;
```

Comment:
put all of this in a finally block, just in case.

Reply:
Done